src/config/loader.ts

```diff
@@ -110,6 +112,19 @@ async function _loadUserConfig(
           configs.rc?.[key] ??
           configs.packageJson?.[key]) as NitroConfig[K];
 
+      const configRootDir =
+        configs.main?.rootDir ??
+        configs.rc?.rootDir ??
+        configs.packageJson?.rootDir;
+      if (
+        configRootDir &&
+        resolve(configRootDir) !== resolve(configOverrides.rootDir || ".")
```

Comment:
We could have a better sanity check [here](https://github.com/kricsleo/nitro/blob/refactor/warn-root-dir/src/config/loader.ts#L78) after resolving everything. `loadedConfig` includes `layers` (and top level layer, if exists should not have `rootDir` with different value